code/math/vecmat.h

```diff
@@ -22,18 +22,32 @@
 //Macros/functions to fill in fields of structures
 //VEC_NULL macros split into two functions in 2009 with commit 75a514b
 
-//macro to check if vector is close to zero or would be close to zero after squaring
+// Null vector checks are performed on the following types of vectors:
+// * orientation component vectors
+// * positions
+// * velocities
+// * normals
+// In each of these cases, FLT_EPSILON or 1.192092896e-07F is a reasonable threshold.
+
+// macro to check if vector is close to zero or would be close to zero after squaring
+// (uses FLT_EPSILON; original threshold was 1e-16 which can be tightened up a bit)
 #define IS_VEC_NULL_SQ_SAFE(v) \
-		(fl_near_zero((v)->xyz.x, (float) 1e-16) && \
-		fl_near_zero((v)->xyz.y, (float) 1e-16) && \
-		fl_near_zero((v)->xyz.z, (float) 1e-16))
-
-//macro to check if vector is close to zero
-#define IS_VEC_NULL(v) \
-		(fl_near_zero((v)->xyz.x, (float) 1e-36) && \
-		fl_near_zero((v)->xyz.y, (float) 1e-36) && \
-		fl_near_zero((v)->xyz.z, (float) 1e-36))
-
+		(fl_near_zero((v)->xyz.x) && \
+		fl_near_zero((v)->xyz.y) && \
+		fl_near_zero((v)->xyz.z))
+
+// macro to check if vector is close to zero
+// (original threshold was 1e-36 which was too small)
+#define IS_VEC_NULL(v) IS_VEC_NULL_SQ_SAFE(v)
+
+// macro to check if moment-of-inertia vector is close to zero
+// (uses different thresholds since MOI values are really small)
+#define IS_MOI_VEC_NULL(v) \
```

Comment:
Why is this `1e-20` now instead of the previous `1e-36`? There's models with MOI values smaller than `1e-20`, and I'm guessing changing this value from anything besides what it was previously will lead to un-wanted/un-needed print statements.

Reply:
I looked at a number of models, including the TSC 2 and the Sathanas, and the smallest magnitude I found was e-19.  Where did you find smaller values?

Reply:
Ah, FotG has some that go as small as `e-24`.

Reply:
I should also mention that I ran through several missions using a test build based on this PR, and every mission functioned properly without throwing any null vector or null MOI warnings.

Reply:
Why not just leave it as-is? What are we gaining from introducing warnings on this data?

Reply:
The point is to determine what a sensible threshold should be.  If FotG has values down to e-24, then that indicates a better threshold is e-25.

Reply:
Based on the way these actual numbers are being used, the main float accuracy concern is that the matrix's determinant still lies within the type's representational bound (since it involves cubes of quantities, this is quite a concern), but since this calculation was changed to use doubles in d92c7a8ea3799adff961d4e3d24f34bdf5f5c5b4, there's no way for this to happen with just floats as input.

Given that, if it can fit in a float, it is theoretically sound. Since this works against forces, the results of using extremely small MOIs tends toward 0, and not something potentially more damaging like infinity.

Reply:
Does this mean that arbitrarily small floating point values can still be valid MOIs?

Or here is another question: do we, in practice, see any MOI matrices where every value is smaller than e-24?

Reply:
There is currently going to be some lower bound on extant pofs, excluding those which are all 0, but I don't think it's right to say 'the limit is whatever the current examples are', the decision should be based on the actual way in which these values are _used_. And the way in which they are used I think allows for very very small values. Subnormal values are where the bad things really start, and that starts at e-38. Given it appears the old value was e-36 that still gives 100x of wiggle room.

TLDR: I think we would be better served by taking the 'floor' and adding padding up from there, than taking a crude estimate of what the current lowest is, and subtracting padding down.

Reply:
Ok, I can go with that.